### PR TITLE
Bootstrap

### PR DESCRIPTION
**Pull request by mbannick**
(this branch is poorly named) -- Bootstrap modules for uncertainty

### CODE REVIEW

File: src/anml/parameter/spline_variable.py [hunk 1]
```diff
@@ -163,7 +163,7 @@ def create_spline(self, df: pd.DataFrame):
             r_linear=self.r_linear
         )
 
-    def _design_matrix(self, df: pd.DataFrame) -> np.ndarray:
+    def _design_matrix(self, df: pd.DataFrame, create_spline: bool = True) -> np.ndarray:
```
Comment (jizezhang) [inline src/anml/parameter/spline_variable.py:166]: is `create_spline` used in the function? and sorry, could you remind me of the reason for adding this argument? I remember we had some discussion on checking whether a spline is created or not, and call the function `create_spline` based on that, but not entirely sure the details... >_< thanks!

Comment (mbannick) [reply src/anml/parameter/spline_variable.py:166]: if we pass in a new data frame, we need to create the spline design matrix, but we don't want to create new knots or anything. If we pass `create_spline=False` then it won't overwrite the original knots and bases

Comment (jizezhang) [reply src/anml/parameter/spline_variable.py:166]: ah yes, that totally makes sense! thanks! so would we want something like
```
if create_spline:
    self.create_spline(df)
```

Comment (mbannick) [reply src/anml/parameter/spline_variable.py:166]: Ah! Yes thanks for catching that. I just pushed a quick fix to master.

**Pull request merged**